python/lsst/ts/tunablelaser/csc.py

```diff
@@ -176,55 +181,67 @@ async def handle_summary_state(self):
         """Handle the summary state transitons."""
         if self.disabled_or_enabled:
             if self.simulation_mode:
-                if self.simulator is None:
-                    self.log.debug("Starting simulator.")
+                if self.model is not None:
+                    self.log.info("Starting simulator.")
                     simulatorcls = getattr(
                         mock_server, f"{type(self.model).__name__}Server"
                     )
                     if not self.unstable:
                         self.simulator = simulatorcls()
                         self.log.debug(f"Chose {self.simulator=}")
                         await self.simulator.start_task
-
+                if self.thermal_ctrl is not None:
                     self.thermal_ctrl_simulator = mock_server.TempCtrlServer(
                         host=self.thermal_ctrl.host
                     )
                     await self.thermal_ctrl_simulator.start_task
                     self.thermal_ctrl.host = self.thermal_ctrl_simulator.host
                     self.thermal_ctrl.port = self.thermal_ctrl_simulator.port
 
-            if not self.connected and self.model is not None:
+            if not self.laser_connected:
                 await self.evt_detailedState.set_write(
                     detailedState=TunableLaser.LaserDetailedState.NONPROPAGATING_CONTINUOUS_MODE
                 )
                 try:
                     await self.model.connect()
                 except Exception:
-                    await self.fault(code=2, report="Connection failed.")
-                    return
-                await self.model.clear_fault()
-                if self.laser_type == "Main":
-                    await self.model.set_optical_configuration(self.optical_alignment)
-                await self.thermal_ctrl.connect()
-            if (
-                self.summary_state == salobj.State.DISABLED
-                and self.model.is_propagating
-            ):
-                await self.model.stop_propagating()
-                await self.publish_new_detailed_state(
-                    TunableLaser.LaserDetailedState.NONPROPAGATING_CONTINUOUS_MODE
-                )
+                    self.log.exception("Connection to laser failed.")
+                    await self.evt_errorCode.set_write(
```

Comment:
you should really not publish this event stand alone like this. This is part of going to fault, which I see you commented below. What is the idea here? you want to publish an error but not go to fault? this is really not allowed.

I think what you might want to do here is expand the error message in the exception above to specify that the laser functionality won't be available. Something like:

```
self.log.exception("Failed to connect to the laser controller. The CSC will operate with reduced functionality.")
```